meson.build

```diff
@@ -5,6 +5,7 @@ project(
         license: 'Apache',
         default_options: [
                 'c_std=c11',
+                'warning_level=3',
```

Comment:
Level `3` contains several warnings that we do not support upstream. We generally only recommend level `2`.

Also, can you explain why you want this here? The warning-levels of meson are not tied to any particular "expressiveness", so we just specify the warning-levels in our CI directly (https://github.com/c-util/automation/blob/master/src/ci-c-util/ci/run-job#L143).